### ISSUE TITLE
BUG: Ensure modules using internal/httpauth retries with correct nonce after 401.

### ISSUE THREAD

**Issue by evanlinjin** [skycoin/skywire-testnet#345]
Log extract:
```
╰─$ go run ./cmd/skywire-node swA.json                                                                                                                                        1 ↵
[2019-05-09T23:46:38+12:00] INFO [pathutil]: using args[0] as config path: swA.json
[2019-05-09T23:46:56+12:00] INFO [messenger]: Opened new channel local ID 3, remote ID 1 with 0348c941c5015a05c455ff238af2e57fb8f914c399aab604e9abb5b32b91a4c1fe
[2019-05-09T23:46:56+12:00] WARN [trmanager]: Failed to accept connection: entry set: status: 401, error: {"error":{"message":"SW-Nonce does not match","code":401}}
```

The client after receiving `{"error":{"message":"SW-Nonce does not match","code":401}}`, must:
1. Grab correct nonce via `(*httpauth.Client).Nonce()`.
2. Retry.

**Comment by evanlinjin** [comment]
@ivcosla this bug is critical, please fix.

![image](https://user-images.githubusercontent.com/14065705/57453674-18adb980-72bb-11e9-8aa3-dd80dc68bd8a.png)

**Comment by evanlinjin** [comment]
```
[2019-05-10T01:21:06+12:00] INFO [messenger]: Updating discovery entry
[2019-05-10T01:21:06+12:00] INFO [trmanager]: Transport 38254102-9309-5c24-85c0-d3098eb81256 failed with error: EOF. Re-dialing...
[2019-05-10T01:21:06+12:00] INFO [trmanager]: Transport 236e33a2-8bc1-5b1d-b168-593931d16111 failed with error: EOF. Re-dialing...
[2019-05-10T01:21:06+12:00] WARN [trmanager]: Failed to change transport status: status: 401, error: {"error":{"message":"SW-Nonce does not match","code":401}}
[2019-05-10T01:21:06+12:00] INFO [trmanager]: De-registered transport 38254102-9309-5c24-85c0-d3098eb81256
[2019-05-10T01:21:06+12:00] WARN [trmanager]: Failed to delete transport: write tcp [::1]:50281->[::1]:8080: use of closed network connection
[2019-05-10T01:21:06+12:00] WARN [trmanager]: Failed to change transport status: status: 401, error: {"error":{"message":"SW-Nonce does not match","code":401}}
[2019-05-10T01:21:06+12:00] INFO [trmanager]: De-registered transport 236e33a2-8bc1-5b1d-b168-593931d16111
[2019-05-10T01:21:06+12:00] WARN [trmanager]: Failed to delete transport: write tcp [::1]:50281->[::1]:8080: use of closed network connection
[2019-05-10T01:21:06+12:00] INFO [trmanager]: Closing transport manager
[2019-05-10T01:21:06+12:00] INFO [trmanager]: Transport a13b7256-8c99-5a47-b82c-982d9edb5442 closed
[2019-05-10T01:21:06+12:00] INFO [trmanager]: Transport a13b7256-8c99-5a47-b82c-982d9edb5442 closed
[2019-05-10T01:21:06+12:00] INFO [trmanager]: Transport a13b7256-8c99-5a47-b82c-982d9edb5442 closed
[2019-05-10T01:21:06+12:00] INFO [trmanager]: Transport a13b7256-8c99-5a47-b82c-982d9edb5442 closed
[2019-05-10T01:21:06+12:00] INFO [trmanager]: Transport 1341640f-0fbb-567f-bac9-4650222e2d87 closed
[2019-05-10T01:21:06+12:00] INFO [trmanager]: Transport a13b7256-8c99-5a47-b82c-982d9edb5442 closed
[2019-05-10T01:21:06+12:00] INFO [trmanager]: Transport a13b7256-8c99-5a47-b82c-982d9edb5442 closed
[2019-05-10T01:21:06+12:00] INFO [trmanager]: Transport a13b7256-8c99-5a47-b82c-982d9edb5442 closed
[2019-05-10T01:21:06+12:00] INFO [trmanager]: Transport a13b7256-8c99-5a47-b82c-982d9edb5442 closed
[2019-05-10T01:21:06+12:00] WARN [trmanager]: Failed to change transport status: status: 401, error: {"error":{"message":"SW-Nonce does not match","code":401}}
```

**Comment by jdknives** [comment]
@evanlinjin this is the bug you suggested @nkryuchkov should tackle?

**Comment by jdknives** [comment]
Closed by #349
